Source/Core/Core/IOS/SDIO/SDIOSlot0.cpp

```diff
@@ -187,22 +208,27 @@ s32 SDIOSlot0::ExecuteCommand(const Request& request, u32 _BufferIn, u32 _Buffer
     break;
 
   case SEND_IF_COND:
+    INFO_LOG(IOS_SD, "SEND_IF_COND");
     // If the card can operate on the supplied voltage, the response echoes back the supply
     // voltage and the check pattern that were set in the command argument.
+    // This command is used to differentiate between protocol v1 and v2.
+    m_protocol = SDProtocol::V2;
     Memory::Write_U32(req.arg, _BufferOut);
     break;
 
   case SEND_CSD:
-    INFO_LOG(IOS_SD, "SEND_CSD");
-    // <WntrMute> shuffle2_, OCR: 0x80ff8000 CID: 0x38a00000 0x480032d5 0x3c608030 0x8803d420
-    // CSD: 0xff928040 0xc93efbcf 0x325f5a83 0x00002600
-
-    // Values used currently are from lpfaint99
-    Memory::Write_U32(0x80168000, _BufferOut);
-    Memory::Write_U32(0xa9ffffff, _BufferOut + 4);
-    Memory::Write_U32(0x325b5a83, _BufferOut + 8);
-    Memory::Write_U32(0x00002e00, _BufferOut + 12);
-    break;
+  {
+    std::array<u32, 4> csd;
+    if (m_protocol == SDProtocol::V1)
+      csd = GetCSDv1();
+    else
+      csd = GetCSDv2();
```

Comment:
`const std::array<u32, 4> csd = m_protocol == SDProtocol::V1 ? GetCSDv1() : GetCSDv2()`. 

As an alternative, it might be a good idea to pass the `m_protocol` parameter to a `GetCSD(SDProtocol protocol)` and work from there.